crates/ra_hir_def/src/adt.rs

```diff
@@ -189,25 +198,41 @@ fn lower_struct(
     match &ast.value {
         ast::StructKind::Tuple(fl) => {
             for (i, fd) in fl.fields().enumerate() {
+                let attrs = Attrs::new(&fd, &Hygiene::new(db.upcast(), ast.file_id));
+
+                // Need verification about parent cfg_options and current with current attributes
+                // If it is we are in a case where the cfg is not enabled then we don't have to add this field to check
+                // if !is_cfg_enabled(&crate_graph[module_id.krate].cfg_options, &attrs) {
+                //     continue;
+                // }
+
                 trace.alloc(
                     || Either::Left(fd.clone()),
                     || StructFieldData {
                         name: Name::new_tuple_field(i),
                         type_ref: TypeRef::from_ast_opt(fd.type_ref()),
                         visibility: RawVisibility::from_ast(db, ast.with_value(fd.visibility())),
+                        attrs: Attrs::new(&fd, &Hygiene::new(db.upcast(), ast.file_id)),
                     },
                 );
             }
             StructKind::Tuple
         }
         ast::StructKind::Record(fl) => {
             for fd in fl.fields() {
+                let attrs = Attrs::new(&fd, &Hygiene::new(db.upcast(), ast.file_id));
+                // Need verification about parent cfg_options and current with current attributes
+                // If it is we are in a case where the cfg is not enabled then we don't have to add this field to check
```

Comment:
Yeah, this is what we *ideally* should do here, but, in the current model, we just have one global set of active cfgs, so there won't be parent in the first place if it is disabled.